src/mixer.cpp

```diff
@@ -36,6 +35,7 @@ struct MixerChannel {
 	bool is16bit;
 };
 
+static std::atomic<uint8> _active_channels;
```

Comment:
`std::atomic` is allowed to use locks under the hood, which might or might not be a bad thing. x86 will probably be lock-less, but e.g. ARM has a lot less interlocked CPU instructions, so might be slightly iffy there.

Reply:
uint8 and indeed anything <= size_t is not a problem on ARM.
ARM (32 bit) does not have separate instructions for each type of atomic operation. ldrex/strex pairs can implement any atomic RMW, broadly equivalent to cmpxchg on x86.

According to godbolt std::atomic<uint8>::is_always_lock_free is true on ARM for GCC and MSVC.

More recent versions of ARMv8 (64 bit) have fancy instructions as well as the above.